master/buildbot/steps/trigger.py

```diff
@@ -199,12 +225,30 @@ def addBuildUrls(self, rclist):
     def run(self):
         schedulers_and_props = yield self.getSchedulersAndProperties()
 
+        schedulers_and_props_list = []
+
+        # To be back compatible we need to differ between old and new style
+        # schedulers_and_props can either consist of 2 elements tuple or dictionary
+        for element in schedulers_and_props:
+            if type(element) == dict:
+                schedulers_and_props_list = schedulers_and_props
+                break
+            else:
+                # Old-style back compatibility: Convert tuple to dict and make it important
+                d = {
+                    'sched_name': element[0],
+                    'props_to_set': element[1],
+                    'unimportant': False
+                }
+                schedulers_and_props_list.append(d)
+
         # post process the schedulernames, and raw properties
         # we do this out of the loop, as this can result in errors
         schedulers_and_props = [(
-            self.getSchedulerByName(sch),
-            self.createTriggerProperties(props_to_set))
-            for sch, props_to_set in schedulers_and_props]
+            self.getSchedulerByName(entry_dict['sched_name']),
+            self.createTriggerProperties(entry_dict['props_to_set']),
+            entry_dict['unimportant'])
```

Comment:
Sorry, my mistake: I misread it as a sequence of method calls, while it's a list comprehension.